nilearn/datasets.py

```diff
@@ -83,27 +83,32 @@ def _chunk_report_(bytes_so_far, total_size, t0):
     bytes_so_far: int
         Number of downloaded bytes
 
-    total_size: int, optional
-        Total size of the file. None is valid
+    total_size: int
+        Total size of the file.
 
-    t0: int, optional
+    t0: int
         The time in seconds (as returned by time.time()) at which the
-        download was started.
+        download was resumed / started.
+
+    initial_size: int
+        If resuming, indicate the initial size of the file.
+        If not resuming, set to zero.
     """
-    if total_size:
-        percent = float(bytes_so_far) / total_size
-        percent = round(percent * 100, 2)
-        dt = time.time() - t0
-        # We use a max to avoid a division by zero
-        remaining = (100. - percent) / max(0.01, percent) * dt
-        # Trailing whitespace is too erase extra char when message length
-        # varies
-        sys.stderr.write(
-            "Downloaded %d of %d bytes (%0.2f%%, %s remaining)  \r"
-            % (bytes_so_far, total_size, percent,
-               _format_time(remaining)))
-    else:
-        sys.stderr.write("Downloaded %d of ? bytes\r" % (bytes_so_far))
+    total_percent = float(bytes_so_far) / total_size
+
+    current_download_size = bytes_so_far - initial_size
+    bytes_remaining = total_size - bytes_so_far
+    dt = time.time() - t0
```

Comment:
If you feel like it, I was planning at some point to change that by the uptime of the machine. otherwise, the estimated time is false if your computer goes to sleep.